File: src/Entity/Conseil.php
<?php

namespace App\Entity;

use App\Entity\User;
use App\Repository\ConseilRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * @ORM\Entity(repositoryClass=ConseilRepository::class)
 */
class Conseil
{


    /**
     * @ORM\Id
     * @Assert\Positive(message="ce champs doit etre positive")
     * @Assert\NotBlank(message="obligatoire")
     * @ORM\Column(type="integer")
     *
     */
    private $reference;

    /**
     *
     * @Assert\NotBlank(message="le nom est obligatoire")
     * @Assert\Regex(pattern="/^[a-z]+$/i", message="ce champs n'accepte pas les chiffres")
     * @ORM\Column(type="string", length=255)
     */
    private $remarques;

    /**
     *
     * @ORM\Column(type="date")
     */
    private $date_red;

    /**
     * @ORM\Column(type="date")
     */
    private $date_limite;

    /**
     * @Assert\NotBlank(message="le nom est obligatoire")
     * @Assert\Regex(pattern="/^[a-z]+$/i", message="ce champs n'accepte pas les chiffres")
     * @ORM\Column(type="string", length=255)
     */
    private $nom_derma;

    /**
     * @Assert\Email(message="saisir votre email correctement")
     * @Assert\NotBlank(message="le nom est obligatoire")
     * 
     * @ORM\Column(type="string", length=255)
     */
    private $email;

    /**
     * @ORM\OneToMany(targetEntity=ConsLike::class, mappedBy="conseil")
     */
    private $likes;

    public function __construct()
    {
        $this->likes = new ArrayCollection();
    }






    public function getReference(): ?int
    {
        return $this->reference;
    }
    public function setReference(int $reference): self
    {
       $this->reference= $reference;
       return $this;
    }


    public function getRemarques(): ?string
    {
        return $this->remarques;
    }

    public function setRemarques(string $remarques): self
    {
        $this->remarques = $remarques;

        return $this;
    }

    public function getDateRed(): ?\DateTimeInterface
    {
        return $this->date_red;
    }

    public function setDateRed(\DateTimeInterface $date_red): self
    {
        $this->date_red = $date_red;

        return $this;
    }

    public function getDateLimite(): ?\DateTimeInterface
    {
        return $this->date_limite;
    }

    public function setDateLimite(\DateTimeInterface $date_limite): self
    {
        $this->date_limite = $date_limite;

        return $this;
    }

    public function getNomDerma(): ?string
    {
        return $this->nom_derma;
    }

    public function setNomDerma(string $nom_derma): self
    {
        $this->nom_derma = $nom_derma;

        return $this;
    }

    public function getEmail(): ?string
    {
        return $this->email;
    }

    public function setEmail(string $email): self
    {
        $this->email = $email;

        return $this;
    }

    /**
     * @return Collection|ConsLike[]
     */
    public function getLikes(): Collection
    {
        return $this->likes;
    }

    public function addLike(ConsLike $like): self
    {
        if (!$this->likes->contains($like)) {
            $this->likes[] = $like;
            $like->setConseil($this);
        }

        return $this;
    }

    public function removeLike(ConsLike $like): self
    {
        if ($this->likes->removeElement($like)) {
            // set the owning side to null (unless already changed)
            if ($like->getConseil() === $this) {
                $like->setConseil(null);
            }
        }

        return $this;
    }


    /**
     * @param \App\Entity\User $user
     * @return bool
     */
    public function islikedByUser(User $user) : bool
    {  foreach ($this->likes as $like)
    {  if ($like->getUser()===$user) return true ;
    }
        return false;
    }
    public function __toString()
    {
        // TODO: Implement __toString() method.
        return (string)$this->getNomDerma();
    }
}

File: sprint java/Entities/Conseil.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Entities;

import java.util.Comparator;
import java.util.Date;
import java.util.Objects;

/**
 *
 * @author amine
 */
public class Conseil implements Comparable <Conseil>  {
    private int reference;
    private String remarques,nom_derma,email;
    private Date date_red,date_limite;

    public Conseil(int reference, String remarques, String nom_derma, String email, Date date_red, Date date_limite) {
       this.reference=reference;
        this.remarques = remarques;
        this.nom_derma = nom_derma;
        this.email = email;
        this.date_red = date_red;
        this.date_limite = date_limite;
    }

    public Conseil() {
        
    }

    public int getReference() {
        return reference;
    }

    public String getRemarques() {
        return remarques;
    }

    public String getNom_derma() {
        return nom_derma;
    }

    public String getEmail() {
        return email;
    }

    public Date getDate_red() {
        return date_red;
    }

    public Date getDate_limite() {
        return date_limite;
    }

    public void setRemarques(String remarques) {
        this.remarques = remarques;
    }

    public void setNom_derma(String nom_derma) {
        this.nom_derma = nom_derma;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public void setDate_red(Date date_red) {
        this.date_red = date_red;
    }

    public void setDate_limite(Date date_limite) {
        this.date_limite = date_limite;
    }

    @Override
    public String toString() {
        return "Conseil{" + "reference=" + reference + ", remarques=" + remarques + ", nom_derma=" + nom_derma + ", email=" + email + ", date_red=" + date_red + ", date_limite=" + date_limite + '}';
    }

    public void setReference(int reference) {
        this.reference = reference;
    }
     public static Comparator <Conseil> ComparatorReference =(Conseil o1, Conseil o2) -> o1.getReference()-o2.getReference();

    @Override
    public int hashCode() {
        int hash = 5;
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Conseil other = (Conseil) obj;
        if (this.reference != other.reference) {
            return false;
        }
        if (!Objects.equals(this.remarques, other.remarques)) {
            return false;
        }
        if (!Objects.equals(this.nom_derma, other.nom_derma)) {
            return false;
        }
        if (!Objects.equals(this.email, other.email)) {
            return false;
        }
        if (!Objects.equals(this.date_red, other.date_red)) {
            return false;
        }
        if (!Objects.equals(this.date_limite, other.date_limite)) {
            return false;
        }
        return true;
    }

    @Override
    public int compareTo(Conseil o) {
         return date_red.compareTo(o.getDate_red());
    }

   
    
    
}
